packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -131,32 +132,45 @@ async function instantiateModule(
   const pendingDeps: string[] = []
 
   const ssrImport = async (dep: string) => {
-    if (dep[0] !== '.' && dep[0] !== '/') {
-      return nodeImport(dep, mod.file!, resolveOptions)
-    }
-    // convert to rollup URL because `pendingImports`, `moduleGraph.urlToModuleMap` requires that
-    dep = unwrapId(dep)
-    if (!isCircular(dep) && !pendingImports.get(dep)?.some(isCircular)) {
-      pendingDeps.push(dep)
-      if (pendingDeps.length === 1) {
-        pendingImports.set(url, pendingDeps)
+    try {
+      if (dep[0] !== '.' && dep[0] !== '/') {
+        return await nodeImport(dep, mod.file!, resolveOptions)
       }
-      const mod = await ssrLoadModule(
-        dep,
-        server,
-        context,
-        urlStack,
-        fixStacktrace,
-      )
-      if (pendingDeps.length === 1) {
-        pendingImports.delete(url)
+      // convert to rollup URL because `pendingImports`, `moduleGraph.urlToModuleMap` requires that
+      dep = unwrapId(dep)
+      if (!isCircular(dep) && !pendingImports.get(dep)?.some(isCircular)) {
+        pendingDeps.push(dep)
+        if (pendingDeps.length === 1) {
+          pendingImports.set(url, pendingDeps)
+        }
+        const mod = await ssrLoadModule(
+          dep,
+          server,
+          context,
+          urlStack,
+          fixStacktrace,
+        )
+        if (pendingDeps.length === 1) {
+          pendingImports.delete(url)
+        } else {
+          pendingDeps.splice(pendingDeps.indexOf(dep), 1)
+        }
+        // return local module to avoid race condition #5470
+        return mod
+      }
+      return moduleGraph.urlToModuleMap.get(dep)?.ssrModule
+    } catch (err) {
+      let errorData = importErrors.get(err)
+      // tell external error handler which mod was imported with error
+      if (errorData) {
+        errorData.importee = dep
       } else {
-        pendingDeps.splice(pendingDeps.indexOf(dep), 1)
+        errorData = { importee: dep }
       }
-      // return local module to avoid race condition #5470
-      return mod
+      importErrors.set(err, errorData)
```

Comment:
Could we simplify this to a single `importErrors.set(err, { importee: dep ))`? Since we would always override it.